src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java

```diff
@@ -11,20 +12,35 @@
 import java.util.List;
 import java.util.Optional;
 
+import static uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.CallbackParams.Params.BEARER_TOKEN;
+
 @Service
 public class CallbackHandlerFactory {
 
+    private final UserService userService;
     private HashMap<String, CallbackHandler> eventHandlers = new HashMap<>();
 
     @Autowired(required = false)
-    public CallbackHandlerFactory(List<CallbackHandler> beans) {
+    public CallbackHandlerFactory(List<CallbackHandler> beans, UserService userService) {
+        this.userService = userService;
         beans.forEach(bean -> bean.register(eventHandlers));
     }
 
     public CallbackResponse dispatch(CallbackParams callbackParams) {
-        return Optional.ofNullable(eventHandlers.get(callbackParams.getRequest().getEventId()))
+        String authorisation = callbackParams.getParams().get(BEARER_TOKEN).toString();
+        String eventId = callbackParams.getRequest().getEventId();
+        return Optional.ofNullable(eventHandlers.get(eventId))
+            .filter(h -> hasSupportedRoles(h, authorisation, eventId))
             .map(h -> h.handle(callbackParams))
-            .orElseThrow(() -> new CallbackException(
-                "Could not handle callback for event " + callbackParams.getRequest().getEventId()));
+            .orElseThrow(() -> new CallbackException("Could not handle callback for event " + eventId));
+    }
+
+    private boolean hasSupportedRoles(CallbackHandler callbackHandler, String authorisation, String eventId) {
+        List<String> userRoles = userService.getUserDetails(authorisation).getRoles();
+        if (callbackHandler.getSupportedRoles().stream().anyMatch(userRoles::contains)) {
+            return true;
+        } else {
+            throw new CallbackException("User does not have supported role for event " + eventId);
```

Comment:
Should have been better to have specific Exception for this and mapped to HTTP Unauthorised response code. How about using `ForbiddenActionException` which is already mapped?

Reply:
done